lib/Backends/OpenCL/OpenCL.cpp

```diff
@@ -561,12 +566,69 @@ void OpenCLFunction::execute() {
 
       if (auto *SI = dyn_cast<SplatInst>(&I)) {
         // Pass the splat as a parameter.
-        setKernelArg(kernel, numArgs + 1, SI->getValue());
+        setKernelArg(kernel, ++numArgs, SI->getValue());
       } else if (auto *EPI = dyn_cast<ElementPowInst>(&I)) {
         // Pass the exp as a parameter.
-        setKernelArg(kernel, numArgs + 1, EPI->getExp());
+        setKernelArg(kernel, ++numArgs, EPI->getExp());
       }
 
+      if (isQuantized) {
+        int32_t destOffset = I.getOperand(0).first->getType()->getOffset();
+        float destScale = I.getOperand(0).first->getType()->getScale();
+        auto lhsScaleParams = quantization::quantizeScaleOffset32To8(
+            I.getOperand(1).first->getType()->getScale() / destScale,
+            I.getOperand(1).first->getType()->getOffset());
+        auto rhsScaleParams = quantization::quantizeScaleOffset32To8(
+            I.getOperand(2).first->getType()->getScale() / destScale,
+            I.getOperand(2).first->getType()->getOffset());
+        setKernelArg(kernel, ++numArgs, destOffset);
+        if (auto *EM = dyn_cast<ElementMulInst>(&I)) {
+          auto mulScaleParams = quantization::quantizeScaleOffset32To8(
+              EM->getLHS()->getType()->getScale() *
+                  EM->getRHS()->getType()->getScale() / destScale,
+              0);
+          setKernelArg(kernel, ++numArgs, lhsScaleParams.offset_);
+          setKernelArg(kernel, ++numArgs, rhsScaleParams.offset_);
+          setKernelArg(kernel, ++numArgs, mulScaleParams);
+        } else {
+          setKernelArg(kernel, ++numArgs, lhsScaleParams);
+          setKernelArg(kernel, ++numArgs, rhsScaleParams);
+        }
+      }
+      enqueueKernel(commands_, kernel, deviceId_, {global}, kernelLaunches_);
+      continue;
+    }
+
+    // Quantize floating point tensor. Scale and Offset are based on return type
+    // of the instruction \p I.
+    if (auto *QI = dyn_cast<QuantizeInst>(&I)) {
```

Comment:
This is fine, but smaller granularity PR is faster to review/ship.
Would be good to keep one quantized operation per PR unless the other one really touches few lines.